File: code/317/solution.cpp
#include <iostream>/*{{{*/
#include <math.h>
using namespace std; /*}}}*/
long double solve(long double hight, long double v, long double g) {
    long double upper = v * v / g / 2 + hight;
    return acos(-1) * upper * upper / g * v * v;
}
int main() {
    printf("%.4Lf\n", solve(100, 20, 9.81));
    return 0;
}

File: code/317/solution.go
package main /*{{{*/

import (
	"fmt"
	"math"
)

func solve(hight, v, g float64) float64 {
	upper := v*v/g/2 + hight
	return math.Acos(-1) * upper * upper / g * v * v
}

func main() {
	fmt.Printf("%.4f\n", solve(100, 20, 9.81))
}
